File: typescript-connector/src/models/ConceptScheme.ts
// Class from DFC Business Ontology: #ConceptScheme
import { SemanticObject } from "../core/SemanticObject.js";

export interface ConceptSchemeParams {
  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;
}
export class ConceptScheme extends SemanticObject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:ConceptScheme";
  }

  date?: string;
  description?: string;
  name?: string;
  characteristicOf?: string;
  hasDimension?: string;

  constructor(
    semanticId: string,
    params?: ConceptSchemeParams,
  ) {
    super(semanticId);
    this.date = params?.date;
    this.description = params?.description;
    this.name = params?.name;
    this.characteristicOf = params?.characteristicOf;
    this.hasDimension = params?.hasDimension;
    this.semanticType = ConceptScheme.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:date", () => this.date);
    this.registerSemanticProperty("dfc-b:description", () => this.description);
    this.registerSemanticProperty("dfc-b:name", () => this.name);
    this.registerSemanticProperty("dfc-b:characteristicOf", () => this.characteristicOf);
    this.registerSemanticProperty("dfc-b:hasDimension", () => this.hasDimension);
  }
  static {
    SemanticObject.typeRegistry.set(ConceptScheme.SEMANTIC_TYPE, ConceptScheme);
  }
}

File: ruby-gem/lib/models/concept_scheme.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #ConceptScheme
require_relative '../core/semantic_object'


module DfcLinkmlConnector
  module Models
    class ConceptScheme < Core::SemanticObject
      SEMANTIC_TYPE = "dfc-b:ConceptScheme".freeze

      # @return [String]
      attr_accessor :date

      # @return [String]
      attr_accessor :description

      # @return [String]
      attr_accessor :name

      # @return [String]
      attr_accessor :characteristic_of

      # @return [String]
      attr_accessor :dimension

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil)
        super(semanticId)
        @date = date
        @description = description
        @name = name
        @characteristic_of = characteristicOf
        @dimension = dimension
        self.semanticType = "dfc-b:ConceptScheme"
        registerSemanticProperty("dfc-b:ConceptScheme:date", &method("date")).valueSetter = method("date=")
        registerSemanticProperty("dfc-b:ConceptScheme:description", &method("description")).valueSetter = method("description=")
        registerSemanticProperty("dfc-b:ConceptScheme:name", &method("name")).valueSetter = method("name=")
        registerSemanticProperty("dfc-b:ConceptScheme:characteristic_of", &method("characteristic_of")).valueSetter = method("characteristic_of=")
        registerSemanticProperty("dfc-b:ConceptScheme:has_dimension", &method("dimension")).valueSetter = method("dimension=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end
